### PR TITLE
Implement MessageDatabase and Firebase corresponding object

### PR DESCRIPTION
**Pull request by MarcelTorne**
This is the first step towards resolving issue #46

### CODE REVIEW

File: app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    override fun onChildAdded(snapshot: DataSnapshot, previousChildName: String?) {
+        val chatMessage = snapshot.getValue(ChatMessage::class.java)
+
+        if (chatMessage != null) {
+            messages.add(chatMessage)
+            this.notifyObservers()
+        }
+    }
+
+    override fun onChildChanged(snapshot: DataSnapshot, previousChildName: String?) {
```
Comment (Bwananah) [inline app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt:44]: Is it necessary to override without replacing it with anything ?

File: app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class FirebaseMessageDatabaseAdapterTest {
+    private val MOCK_PATH = "POI/mock-poi"
+    private lateinit var db: FirebaseMessageDatabaseAdapter
+    private val fake_message =
+        ChatMessage("fake_id", "fake_text", "fake_from_id", "fake_to_id", 12345)
+
+    @Before
+    fun initializeDatabase() {
+        UiThreadStatement.runOnUiThread {
+            db = FirebaseMessageDatabaseAdapter(MOCK_PATH)
```
Comment (Bwananah) [inline app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt:22]: What is MOCK_PATH ? Is it used to create a mock database ?

**Review by Bwananah — COMMENTED** [review]
Very good job. You are a machine ! I would prefer if you added more comments to explain your code as I am not as familiar with android or Firebase. But I should get better at reading code as the semester continues.

I am waiting until you see my questions before I approve, but it should be fine

File: app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt (hunk 1 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapter.kt:44]: Unfortunately yes it is

File: app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt (hunk 2 above)
Comment (MarcelTorne) [reply app/src/androidTest/java/com/github/epfl/meili/messages/FirebaseMessageDatabaseAdapterTest.kt:22]: I will write it as comment, path is the path of the Firebase database, so yes it is for a mock database

**Review by Bwananah — APPROVED** [review]
Thanks for the changes, good job !

**Review by ahmedc3ca — APPROVED** [review]
A great addition to the code. I will need to rebase this to my branch to use the firebaseMessageAdapter.

**Pull request merged**